openhcl/underhill_core/src/loader/mod.rs

```diff
@@ -416,61 +416,97 @@ pub fn write_uefi_config(
         entropy
     }));
 
-    let acpi_builder = AcpiTablesBuilder {
-        processor_topology,
-        mem_layout,
-        cache_topology: None,
-        with_ioapic: cfg!(guest_arch = "x86_64"), // underhill always runs with ioapic on x64
-        with_pic: false,                          // uefi never runs with pic or pit
-        with_pit: false,
-        with_psp: platform_config.general.psp_enabled,
-        pm_base: crate::worker::PM_BASE,
-        acpi_irq: crate::worker::SYSTEM_IRQ_ACPI,
-    };
+    // We will generate these tables unless trusted tables are passed via DevicePlatformSettings
+    let mut build_madt = true;
```

Comment:
I actually think this bool-driven flow is quite reasonable, at least, when we're only dealing with these 2 tables.